hypertrace-view-generator/hypertrace-view-generator/src/main/java/org/hypertrace/viewgenerator/generators/SpanEventViewGenerator.java

```diff
@@ -140,21 +143,93 @@ private Event getApiEntrySpanForExitSpan(
-      apiNode
-          .getEvents()
-          .forEach(e -> eventToApiExitCallCount.put(e.getEventId(), totalExitCallCount));
+      ApiExitCallInfo apiExitCallInfo = new ApiExitCallInfo().withExitCallCount(totalExitCallCount);
+      edges.forEach(edge -> apiExitCallInfo.handleApiNodeEdge(trace, edge));
+      backendExitEvents.forEach(apiExitCallInfo::handleBackend);
+      apiNode.getEvents().forEach(e -> eventToExitInfo.put(e.getEventId(), apiExitCallInfo));
+    }
+    return eventToExitInfo;
+  }
+
+  static class ApiExitCallInfo {
+
+    private static final String UNKNOWN_SERVICE = "unknown-service";
+    private static final String UNKNOWN_BACKEND = "unknown-backend";
+
+    private final Map<String, String> calleeIdToName;
+    private final Map<String, AtomicInteger> calleeIdToCallCount;
+
+    private int exitCallCount;
+    private int unknownServiceExits;
+    private int unknownBackendExits;
+
+    public ApiExitCallInfo() {
+      unknownServiceExits = 0;
+      unknownBackendExits = 0;
+      this.calleeIdToName = Maps.newHashMap();
+      this.calleeIdToCallCount = Maps.newHashMap();
+    }
+
+    public ApiExitCallInfo withExitCallCount(int exitCallCount) {
+      this.exitCallCount = exitCallCount;
+      return this;
+    }
+
+    void handleApiNodeEdge(StructuredTrace trace, ApiNodeEventEdge edge) {
+      Event event = trace.getEventList().get(edge.getTgtEventIndex());
+      if (EnrichedSpanUtils.containsServiceId(event)) {
+        calleeIdToName.put(
+            EnrichedSpanUtils.getServiceId(event), EnrichedSpanUtils.getServiceName(event));
+        calleeIdToCallCount
+            .computeIfAbsent(EnrichedSpanUtils.getServiceId(event), v -> new AtomicInteger(0))
+            .incrementAndGet();
+      } else {
+        unknownServiceExits++;
+      }
+    }
+
+    void handleBackend(Event exitEvent) {
+      if (exitEvent
+          .getEnrichedAttributes()
+          .getAttributeMap()
+          .containsKey(EntityConstants.getValue(BackendAttribute.BACKEND_ATTRIBUTE_ID))) {
+        calleeIdToName.put(
+            EnrichedSpanUtils.getBackendId(exitEvent), EnrichedSpanUtils.getBackendName(exitEvent));
+        calleeIdToCallCount
+            .computeIfAbsent(EnrichedSpanUtils.getBackendId(exitEvent), v -> new AtomicInteger(0))
+            .incrementAndGet();
+      } else {
+        unknownBackendExits++;
+      }
+    }
+
+    int getExitCallCount() {
+      return exitCallCount;
+    }
+
+    Map<String, Integer> getServiceNameToExitCalls() {
+      Map<String, Integer> serviceNameToExitCalls =
+          calleeIdToCallCount.entrySet().stream()
+              .collect(
+                  Collectors.toMap(k -> calleeIdToName.get(k.getKey()), v -> v.getValue().get()));
+      if (unknownServiceExits > 0) {
+        serviceNameToExitCalls.put(UNKNOWN_SERVICE, unknownServiceExits);
+      }
+      if (unknownBackendExits > 0) {
+        serviceNameToExitCalls.put(UNKNOWN_BACKEND, unknownBackendExits);
```

Comment:
This looks weird. `serviceNameToExitCalls` has `UNKNOWN_BACKEND`?

Reply:
Updated the name.